Realm/RLMApp.mm

```diff
@@ -0,0 +1,178 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import <Foundation/Foundation.h>
+#import "RLMApp_Private.hpp"
+
+#import "RLMAppCredentials_Private.hpp"
+#import "RLMSyncUser_Private.hpp"
+#import "RLMSyncManager_Private.h"
+
+using namespace realm;
+
+@implementation RLMAppConfiguration
+
+- (instancetype)initWithBaseURL:(NSString *)baseURL
+                      transport:(id<RLMNetworkTransporting>)transport
+                   localAppName:(NSString *)localAppName
+                localAppVersion:(NSString *)localAppVersion {
+    return [self initWithBaseURL:baseURL
+                       transport:transport
+                    localAppName:localAppName
+                 localAppVersion:localAppVersion
+         defaultRequestTimeoutMS:6000];
+}
+
+-(instancetype) initWithBaseURL:(NSString * _Nullable) baseURL
+                      transport:(id<RLMNetworkTransporting> _Nullable)transport
+                   localAppName:(NSString * _Nullable)localAppName
+                localAppVersion:(NSString * _Nullable)localAppVersion
+        defaultRequestTimeoutMS:(NSUInteger)defaultRequestTimeoutMS {
+    if (self = [super init]) {
+        self.baseURL = baseURL;
+        self.transport = transport;
+        self.localAppName = localAppName;
+        self.localAppVersion = localAppVersion;
+        self.defaultRequestTimeoutMS = defaultRequestTimeoutMS;
+        return self;
+    }
+    return nil;
+}
+@end
+
+/// A map of ObjectStore HTTP methods to RLMHTTPMethods
+std::map<app::HttpMethod, RLMHTTPMethod> HttpMethod_toRLMHTTPMethod = {
```

Comment:
Rather than building a lookup table and translating at runtime, we normally just give the two enums the same values and cast between them, with `static_assert()`s verifying that the values actually are the same. See RLMPropertyType/realm::PropertyType for an example of this (RLMConstants.h and RLMProperty.mm have the relevant bits).